training/creating_dataset.py

```diff
@@ -15,117 +13,102 @@
 
 
 def is_path_correct(path: str) -> bool:
     """
     Checks whether given path is correct: file exists and correct file extension - json or db
+
     :param path: path to book to load
-    :return: True if path is correct otherwise False
+    :return: True if path is correct
     """
     if not os.path.isfile(path):
-        print(f"Path {path} does not exist")
-        return False
+        raise ValueError(f"Path {path} does not exist")
+        # return False
     if not (path.endswith(".db") or path.endswith(".json")):
-        print(f"Expected db or json file, got {path}")
-        return False
+        raise ValueError(f"Expected db or json file, got {path}")
+        # return False
     return True
 
 
-def move_values_level_up(value_name: str) -> callable:
-    """
-    Assigns values of chosen key in columns' dictionaries to that columns
-    :param value_name: name of value that should be moved one level up as value of current column/columns
-    :return: function which create DataFrame with values of current column(s) changed to value of chosen subcolumn
+def load_and_preprocess_data(
+    paths_to_books: list[str],
+    characteristics: list[str],
+) -> pd.DataFrame:
     """
+    Creates dataframe containing chosen characteristics, level and source book of monsters from chosen books
 
-    def inner_move_values_level_up(df: pd.DataFrame) -> pd.DataFrame:
-        return df.applymap(lambda x: x.get(value_name), na_action="ignore")
-
-    return inner_move_values_level_up
-
-
-def get_subcolumn(book: pd.DataFrame, subcolumn_path: str) -> pd.DataFrame:
-    """
-    Gets subcolumn of given DataFrame according to given path
-    :param book: DataFrame with all data from book(s)
-    :param subcolumn_path: path to subcolumn
-    :return: chosen subcolumn as DataFrame
+    :param paths_to_books: list of paths of books to load
+    :param characteristics: list of characteristics to load
+    :return: DataFrame with monsters (NPC) from chosen books and with chosen characteristics and their origin book
     """
-    subcol = deepcopy(book)
-    for col in subcolumn_path.split("/"):
-        subcol = pd.DataFrame(data=unpack_column(subcol, col))
+    data = []
 
-    return subcol
+    for path in paths_to_books:
+        is_path_correct(path)
+        with open(path) as file:
+            # loading json strings from files
+            data += [json.loads(line) for line in file]
+            file.close()
 
+    bestiary = pd.json_normalize(data)
+    # only npc monsters
+    bestiary = bestiary[bestiary["type"] == "npc"]
+    # only system column (all characteristics are there)
+    bestiary = bestiary.filter(regex="system", axis=1)
+
+    columns = []
+    for characteristic in characteristics + ["book", "level"]:
+        # different way of loading data: abilities and saves is a group of characteristics
+        if characteristic in ["abilities", "saves"]:
+            subcolumns = bestiary.filter(regex=characteristic, axis=1)
```

Comment:
Polecam jako axis używać wszędzie "index" albo "columns", jest czytelniejsze. 0 i 1 to taka bardziej zaszłość historyczna